src/constants/oracles.ts

```diff
@@ -0,0 +1,68 @@
+import { ChainId } from '../crypto/networks';
+import { PartialRecord } from '../types/common';
+import { getOneTokenFactoryContract } from '../utils/contracts';
+import { getProvider } from '../crypto/providers';
+import { OracleName, OracleNames } from '../enums/oracleName';
+import { AddressName } from '../enums/addressName';
+import { getAddress } from './addresses';
+import { OneTokenFactory } from '../generated';
+
+type OracleMapping = {
+  [oracleValues in OracleNames]: PartialRecord<ChainId, string>;
+};
+
+// TODO: I think we can totally remove this and merge it into the TOKENS?  So here are some overlap here, really we want
+// tokens to be in tokens.ts and other contracts are fine to be in here...
+export const ORACLES: OracleMapping = {
+  [OracleName.MAINNET_ICHI_ORACLE]: { [ChainId.Mainnet]: '0xD41EA28e17BD06136c416cA942fB997122138139' },
+};
+
+export async function getAllOneTokenOracles(chainId: ChainId): Promise<Map<string, string[]>> {
```

Comment:
A few issues here:

- Get all one token oracles from where, from what?  I say that as you may want to consider forcing the user to provide the one token factory address instead of assuming it.  What if there is another one token factory in the future?  
- Don't return a map, return a `Record<string, string[]>` mainly to be consistent with other code that uses a Record instead of Map